doc/tutorials/learning_to_rank.rst

```diff
@@ -0,0 +1,177 @@
+  n_query_groups = 3
+  qid = rng.integers(0, 3, size=X.shape[0])
+
+  # Sort the inputs based on query index
+  sorted_idx = np.argsort(qid)
+  X = X[sorted_idx, :]
+  y = y[sorted_idx]
+
+The simpliest way to train a ranking model is by using the sklearn estimator interface. Continuing the previous snippet, we can train a simple ranking model without tuning:
+
+.. code-block:: python
+
+  ranker = xgb.XGBRanker(tree_method="hist", lambdarank_num_pair_per_sample=8, objective="rank:ndcg", lambdarank_pair_method="topk")
+  ranker.fit(X, y, qid=qid)
+
+Please note that, as of writing, there's no learning to rank interface in sklearn. As a result, the :py:class:`xgboost.XGBRanker` does not fully conform the sklearn estimator guideline and can not be directly used with some of its utility functions. For instances, the ``auc_score`` and ``ndcg_score`` in sklearn don't consider group information nor the pairwise loss. Most of the metrics are implemented as part of XGBoost, but to use sklearn utilities like :py:func:`sklearn.model_selection.cross_validation`, we need to make some adjustments in order to pass the `qid` as an additional parameter for :py:meth:`xgboost.XGBRanker.score`. The `X` for :py:class:`xgboost.XGBRanker` may contain a special column called ``qid`` when it's a pandas dataframe or a cuDF dataframe:
+
+.. code-block:: python
+
+  df = pd.DataFrame(X, columns=[str(i) for i in range(X.shape[1]))
+  df["qid"] = qid
+  ranker.fit(df, y)  # No need to pass qid as a separate argument
+
+  from sklearn.model_selection import StratifiedGroupKFold, cross_val_score
+  # Works with cv in sklearn, along with HPO utilities like grid search cv.
+  kfold = StratifiedGroupKFold(shuffle=False)
+  cross_val_score(ranker, df, y, cv=kfold, groups=df.qid)
+
+The above snippets build a model using ``LambdaMART`` with the ``NDCG@8`` metric. The outputs of a ranker are relevance scores:
+
+.. code-block:: python
+
+  scores = ranker.predict(X)
+  sorted_idx = np.argsort(scores)[::-1]
+  # Sort the relevance scores from most relevant to least relevant
+  scores = scores[sorted_idx]
+
+
+*************
+Position Bias
+*************
+Real relevance degree for query result is difficult to obtain as it often requires human judegs to examine the content of query results. When such labeled data is absent, we might want to train the model on ground truth data like user clicks. Another upside of using click data directly is that it can relect the up-to-date relevance status `[1] <#references>`__. However, user clicks are often nosiy and biased as users tend to choose results displayed in higher position. To ameliorate this issue, XGBoost implements the ``Unbiased LambdaMART`` `[4] <#references>`__ algorithm to debias the position-dependent click data. The feature can be enabled by the ``lambdarank_unbiased`` parameter, see :ref:`ltr-param` for related options and :ref:`sphx_glr_python_examples_learning_to_rank.py` for a worked example with simulated user clicks.
```

Comment:
```suggestion
Obtaining real relevance degrees for query results is an expensive and strenuous task, as it requires human labelers to label all results one by one. When such labeling task is infeasible, we might want to train the learning-to-rank model on user click data instead, as it is relatively easy to collect. Another advantage of using click data directly is that it can reflect the most up-to-date user preferences `[1] <#references>`__. However, user clicks are often biased,  as users tend to choose results that are displayed in higher positions. User clicks are also noisy, where users accidentally click on irrelevant documents. To ameliorate this issue, XGBoost implements the ``Unbiased LambdaMART`` `[4] <#references>`__ algorithm to debias the position-dependent click data. The feature can be enabled by the ``lambdarank_unbiased`` parameter; see :ref:`ltr-param` for related options and :ref:`sphx_glr_python_examples_learning_to_rank.py` for a worked example with simulated user clicks.
```
Bunch of style improvements. Also treated bias and noise as distinct issues.

Reply:
Maybe I should mention other bias as well, for instance, presentation bias.